optuna/integration/chainermn.py

```diff
@@ -178,6 +182,105 @@ def prune(self, storage, study_id, trial_id, step):
             return result
 
 
+class _ChainerMNTrial(Trial):
+    def __init__(self, trial, comm):
+        # type: (Trial, CommunicatorBase) -> None
+
```

Comment:
Since this class inherits `Trial`,  it seems necessary to call `super(...).__init__(trial.study, trial.trial_id)`.
But I'm wondering that, in this case, it may be sufficient to inherit `BaseTrial` instead of `Trial`.

Reply:
Thank you for your comment. As you mentioned, we can inherit `BaseTrial`, but we also need to change the type of objective functions  from `Callable[[Trial, CommunicatorBase]` to `Callable[[BaseTrial, CommunicatorBase]`. This change will cause the inconsistency of API between `Study.optimize` and `ChainerMNStudy.optimize`, and I want to separate this API change from this PR. What do you think of it?

Reply:
I see, thanks. I agree with your opinion. It seems better to separate the API change to another PR.

Reply:
Thank you for your reply. I just added the `__init__()` call of the super class.